test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
 
 'use strict';
 
-var assert = require('assert');
-var common = require('@google-cloud/common');
-var delay = require('delay');
-var events = require('events');
-var extend = require('extend');
-var PQueue = require('p-queue');
-var proxyquire = require('proxyquire');
-var stackTrace = require('stack-trace');
-var through = require('through2');
-var timeSpan = require('time-span');
-
-var pQueueOverride = null;
-function FakePQueue(options) {
-  return new (pQueueOverride || PQueue)(options);
-}
-
-var fakeStackTrace = extend({}, stackTrace);
+const assert = require('assert');
+const common = require('@google-cloud/common');
+const extend = require('extend');
+const through = require('through2');
 
 describe('SessionPool', function() {
-  var SessionPool;
-  var sessionPool;
+  let SessionPool;
 
-  var DATABASE = {
+  let DATABASE = {
     request: common.util.noop,
     requestStream: common.util.noop,
   };
 
   before(function() {
-    SessionPool = proxyquire('../src/session-pool.js', {
-      'p-queue': FakePQueue,
-      'stack-trace': fakeStackTrace,
-    });
-  });
-
-  beforeEach(function() {
-    sessionPool = new SessionPool(DATABASE);
-    sessionPool.onClose_ = new Promise(function() {});
+    SessionPool = require('../src/session-pool.js');
   });
 
-  afterEach(function() {
-    pQueueOverride = null;
-    fakeStackTrace.get = stackTrace.get;
-  });
+  describe('tests with no min sessions sessionPool before hook', function() {
```

Comment:
I'm not entirely sure I understand what this means, what is this block of tests trying to do here?

Reply:
updated the description